src/kmeans.jl

```diff
@@ -64,27 +63,27 @@ end
 
 # core k-means skeleton
 function _kmeans!(
-    x::Matrix{T},                   # in: sample matrix (d x n)
-    w::Union{Nothing, Vector{T}},      # in: sample weights (n)
-    centers::Matrix{T},             # in/out: matrix of centers (d x k)
-    assignments::Vector{Int},       # out: vector of assignments (n)
-    costs::Vector{T},               # out: costs of the resultant assignments (n)
-    counts::Vector{Int},            # out: the number of samples assigned to each cluster (k)
-    cweights::Vector{Float64},      # out: the weights of each cluster
-    maxiter::Int,                   # in: maximum number of iterations
-    tol::Real,                      # in: tolerance of change at convergence
-    displevel::Int,                 # in: the level of display
-    distance::SemiMetric) where T<:AbstractFloat             # in: function to calculate the distance with
+    X::AbstractMatrix{T},         # in: sample matrix (d x n)
+    w::Union{Nothing, AbstractVector{<:Real}}, # in: sample weights (n)
+    centers::AbstractMatrix{T},   # in/out: matrix of centers (d x k)
+    assignments::Vector{Int},     # out: vector of assignments (n)
+    costs::Vector{T},             # out: costs of the resultant assignments (n)
+    counts::Vector{Int},          # out: # samples assigned to each cluster (k)
+    cweights::Vector{Float64},    # out: weights of each cluster
+    maxiter::Integer,             # in: maximum number of iterations
+    tol::Real,                    # in: tolerance of change at convergence
+    displevel::Int,               # in: the level of display
+    distance::SemiMetric          # in: function to calculate the distance with
+    ) where T<:Real
```

Comment:
The pre-PR code contained both indented and unindented examples of a stray closing bracket + type constraints declaration.
I like unindented version more as it helps to separate the function arguments from its body. @nalimilan What is the recommended convention?

```suggestion
) where T<:Real
```

Reply:
to be honest it's not really like the current code (now and before) is super consistent in terms of style... but can we work on this in another PR potentially with broadcasting and type fix?
To be fair, I'm considering rewriting the whole backbone of this `kmeans.jl` file so that it can use more than one algorithm and can take a `Table` instead of just matrices... style questions and better docstrings could go in there too.

Reply:
I don't think there's a convention about this. In general the Julia code base doesn't use that style, but instead puts the first argument on the same line as the function name.